libs/s25main/buildings/nobMilitary.h

```diff
@@ -138,6 +140,10 @@ class nobMilitary : public nobBaseMilitary
     auto GetTroops() const { return helpers::nonNullPtrSpan(troops); }
     bool IsInTroops(const nofPassiveSoldier& soldier) const;
 
+    const unsigned* GetDesiredTroopsPointer(unsigned rank) const { return &desired_troops[rank]; }
+    unsigned GetDesiredTroops(const unsigned rank) const { return desired_troops[rank]; }
+    void SetDesiredTroops(const unsigned rank, const unsigned count);
```

Comment:
`const not required on declaration:
```suggestion
    void SetDesiredTroops(unsigned rank, unsigned count);
```

Also please add a docstring, especially what "desired" means here exactly